examples/example/App.tsx

```diff
@@ -137,6 +142,39 @@ export type TestElement = MyViewElement extends never ? never : 'ok'
 // should not have TS errors because element type should not be never ✅
 export const testElement: { el: TestElement } = { el: 'ok' }
 
+export const ExtendedComp = styled(KeyboardAvoidingView)({})
+
+export const ExtendedCompWithExtraProps = styled<
+  React.ComponentProps<typeof KeyboardAvoidingView> & { custom?: boolean }
+>(KeyboardAvoidingView)({})
+
+export const ExtendedCompWithExtraProps2 = styled(KeyboardAvoidingView)<
+  React.ComponentProps<typeof KeyboardAvoidingView> & { custom?: boolean }
+>({})
+
+/* @ts-expect-error behavior cannot be 'test' */
+export const ExtendA = () => <ExtendedComp behavior="test" />
+/* should not error */
+export const ExtendB = () => <ExtendedComp behavior="padding" />
+/* @ts-expect-error custom does not exist */
+export const ExtendC = () => <ExtendedComp custom="test" />
+/* @ts-expect-error behavior cannot be 'test' */
+export const ExtendE = () => <ExtendedCompWithExtraProps behavior="test" />
+/* should not error */
+export const ExtendF = () => <ExtendedCompWithExtraProps behavior="padding" />
+/* @ts-expect-error custom cannot be string */
+export const ExtendG = () => <ExtendedCompWithExtraProps custom="test" />
+/* should not error */
+export const ExtendH = () => <ExtendedCompWithExtraProps custom={false} />
+/* @ts-expect-error behavior cannot be 'test' */
+export const ExtendI = () => <ExtendedCompWithExtraProps2 behavior="test" />
+/* should not error */
+export const ExtendJ = () => <ExtendedCompWithExtraProps2 behavior="padding" />
+/* @ts-expect-error custom cannot be string */
+export const ExtendK = () => <ExtendedCompWithExtraProps2 custom="test" />
+/* should not error */
+export const ExtendL = () => <ExtendedCompWithExtraProps2 custom={false} />
```

Comment:
i feel like all this type testing should belong somewhere else other than `examples/example/App.tsx`... some kind of `tests.tsx` file somewhere. let me know if you'd like me to move it all.

Reply:
yeah this should get cleaned up at some point. `examples` isn't a great name for this, it's really just for local dev testing. i point people to the solito [starter](https://solito.dev/starter) instead for using dripsy